### PR TITLE
Feature/#1

### LINKED ISSUE #1: Build workflow
**AS** DEVOPS

**I WANT** to have a way to build and deliver the CI/CD workflow

**SO THAT** when I push a change the version could be built and deployed automatically

**Describe Acceptances Criteria**

> Acceptance criteria are a set of statements, each with a clear pass/fail result, that can be measured and specify both functional and non-functional requirements.

Include a list with the acceptances criteria to give Ok the issue

Given/When/Then Acceptance Criteria is a scenario-oriented criteria format. The format here is – “Scenario: (description of the scenario). Given (what all are the given conditions), when (a particular action taken), then (the result of that action). 

- [x] ***SCENARIO-01*** Build when changes are Pushed on the features branch
- [x] ***SCENARIO-02*** Build when changes are published on the main branch
- [x] ***SCENARIO-03*** Build the release delivery and Publish on releases
- [ ] ***SCENARIO-04*** Measure the Coverage of test on main and feature branches

### PR DESCRIPTION
**Pull request by jbenaventem**
Include Coverage into

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,98 @@
...
+
+```mermaid
+stateDiagram
+    direction LR
+    [*] --> checkout@v3
+    checkout@v3 --> setup_go@v3
+    setup_go@v3 --> vet
+    vet --> fmt
+    fmt --> build
+    build --> [*]
```
Comment (benauca) [inline README.md:48]: Include the coverage actions in the mermaid models

File: coverage [hunk 2]
```diff
@@ -0,0 +1,13 @@
+mode: set
```
Comment (benauca) [inline coverage:1]: Remove this file

**Review by benauca — CHANGES_REQUESTED** [review]

File: coverage (hunk 2 above)
Comment (jbenaventem) [reply coverage:1]: Removed!

**Review by benauca — APPROVED** [review]
LGTM!!
Good Jobs ;)

**Pull request merged**